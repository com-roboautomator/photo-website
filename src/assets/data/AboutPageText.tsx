export const text = `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Eu scelerisque felis imperdiet proin. Pretium aenean pharetra magna ac. Scelerisque fermentum dui faucibus in ornare quam viverra orci. Id porta nibh venenatis cras sed felis eget. Pharetra pharetra massa massa ultricies. Metus dictum at tempor commodo ullamcorper. Leo urna molestie at elementum eu facilisis. Faucibus turpis in eu mi bibendum neque egestas congue. Augue ut lectus arcu bibendum at. Sed adipiscing diam donec adipiscing tristique risus. Dignissim cras tincidunt lobortis feugiat vivamus at augue eget. Ullamcorper eget nulla facilisi etiam dignissim. Rutrum quisque non tellus orci ac auctor. Massa tincidunt dui ut ornare lectus sit amet. Blandit aliquam etiam erat velit scelerisque in dictum non. Eleifend mi in nulla posuere sollicitudin.

Purus sit amet luctus venenatis lectus magna. A diam sollicitudin tempor id eu nisl nunc mi. Mattis molestie a iaculis at. Sed odio morbi quis commodo odio aenean sed adipiscing. Sed vulputate mi sit amet mauris commodo. Habitant morbi tristique senectus et netus. Donec adipiscing tristique risus nec feugiat in. Egestas sed sed risus pretium quam vulputate dignissim. Orci dapibus ultrices in iaculis. Scelerisque mauris pellentesque pulvinar pellentesque habitant morbi tristique senectus et. Elementum nibh tellus molestie nunc non.

Tellus rutrum tellus pellentesque eu tincidunt tortor. Pellentesque id nibh tortor id aliquet lectus proin nibh. Enim ut tellus elementum sagittis vitae et leo. In ornare quam viverra orci sagittis eu volutpat odio facilisis. Eget nunc scelerisque viverra mauris in aliquam sem fringilla ut. Feugiat in ante metus dictum at tempor. Nullam vehicula ipsum a arcu cursus vitae congue. Eget duis at tellus at urna condimentum mattis pellentesque id. Enim ut sem viverra aliquet eget sit. Faucibus interdum posuere lorem ipsum dolor sit amet. Sit amet nisl purus in. Egestas congue quisque egestas diam in arcu.`
