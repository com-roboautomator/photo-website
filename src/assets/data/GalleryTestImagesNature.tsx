import Collection from "./ImageDataStructure";
import { v4 as uuid } from 'uuid'

export const natureImages: Collection[] = [
    {
        key: '017a317f-8285-4cfb-a4e7-668d85721b76',
        title: 'Animal Collection',
        titleImage: 0,
        tagTitle: 'Nature',
        tagColour: '#47742c',
        index: 0,

        images: [
            {
                key: uuid(),
                url: "https://picsum.photos/id/1003/500",
                title: "Deer",
                index: 0
            },
            {
                key: uuid(),
                url: "https://picsum.photos/id/1020/500",
                title: "Bear & Cub",
                index: 1
            },
            {
                key: uuid(),
                url: "https://picsum.photos/id/1024/500",
                title: "Buzzard",
                index: 2
            },
            {
                key: uuid(),
                url: "https://picsum.photos/id/1025/500",
                title: "Pug",
                index: 3
            },
            {
                key: uuid(),
                url: "https://picsum.photos/id/1074/500",
                title: "Lion",
                index: 4
            },
            {
                key: uuid(),
                url: "https://picsum.photos/id/1084/500",
                title: "Walrus",
                index: 5
            },
            {
                key: uuid(),
                url: "https://picsum.photos/id/169/500",
                title: "Dogs",
                index: 6
            },
            {
                key: uuid(),
                url: "https://picsum.photos/id/200/500",
                title: "Moo Cow",
                index: 7
            },
            {
                key: uuid(),
                url: "https://picsum.photos/id/237/500",
                title: "Dog",
                index: 8
            },
            {
                key: uuid(),
                url: "https://picsum.photos/id/244/500",
                title: "Heron",
                index: 9
            },

        ]
    },

    {
        key: 'b97bf8c2-6ab5-45e5-b363-dace717b8ed5',
        title: 'Landscape Collection',
        titleImage: 1,
        tagTitle: 'Nature',
        tagColour: '#47742c',
        index: 1,

        images: [
            {
                key: uuid(),
                url: "https://picsum.photos/id/256/500",
                title: "Landscape #1",
                index: 0
            },
            {
                key: uuid(),
                url: "https://picsum.photos/id/261/500",
                title: "Sand",
                index: 1
            },
            {
                key: uuid(),
                url: "https://picsum.photos/id/277/500",
                title: "Grass",
                index: 2
            },
        ]
    },

    {
        key: '12352dd6-255c-4e38-9230-a96723bf9c5d',
        title: 'Romance Collection',
        tagTitle: 'Life',
        tagColour: '#7baab5',
        index: 2,

        images: [
            {
                key: uuid(),
                url: "https://picsum.photos/id/1004/500",
                title: "People",
                index: 0
            },
            {
                key: uuid(),
                url: "https://picsum.photos/id/336/500",
                title: "Locks",
                index: 1
            },
        ]
    },

    {
        key: 'ffdf5e29-408c-44e9-a453-78ba0adb91f5',
        title: 'Extreme Sporting Collection',
        tagTitle: 'Life',
        tagColour: '#7baab5',
        index: 3,

        images: [

            {
                key: uuid(),
                url: "https://picsum.photos/id/328/500",
                title: "Skiing",
                index: 0
            },
            {
                key: uuid(),
                url: "https://picsum.photos/id/386/500",
                title: "Hiking",
                index: 1
            },

        ]
    },

    {
        key: 'a2cb8bb1-0d7e-445d-b977-be97c616aed3',
        title: 'Party Collection',
        tagTitle: 'Life',
        tagColour: '#7baab5',
        index: 4,

        images: [
            {
                key: uuid(),
                url: "https://picsum.photos/id/158/500",
                title: "Party",
                index: 0
            },
            {
                key: uuid(),
                url: "https://picsum.photos/id/390/500",
                title: "Jumping",
                index: 1
            },
        ]
    },

]
