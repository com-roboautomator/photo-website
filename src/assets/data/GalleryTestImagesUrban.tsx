import collection from "./ImageDataStructure";
import {v4 as uuid} from 'uuid'

export const urbanImages: collection[] = [
    {
        key: uuid(),
        url:
            'https://picsum.photos/id/101/300/300', // won't be required anymore
        titleImage: 4,    
        title: 'Buildings Collection',
        tagTitle: 'Architecture',
        tagColour: '#5f5f5f',
        index: 0,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1031/300/300',
                title: 'Skyscraper',
                index: 0,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1029/300/300',
                title: 'New York',
                index: 1,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1040/300/300',
                title: 'Castle',
                index: 2,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1047/300/300',
                title: 'Alley',
                index: 3,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1067/300/300',
                title: 'City View',
                index: 4,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/164/300/300',
                title: 'Lake Side View',
                index: 5,
            },
        ]
    },

    {
        key: uuid(),
        url:
            'https://picsum.photos/id/1026/300/300',
        title: 'Transport Collection',
        tagTitle: 'Cityscapes',
        tagColour: '#7baab5',
        index: 1,
    },

    {
        key: uuid(),
        url:
            'https://picsum.photos/id/1040/300/300',
        title: 'Historic Building Collection',
        tagTitle: 'Architecture',
        tagColour: '#5f5f5f',
        index: 2,
    },

    {
        key: uuid(),
        url:
            'https://picsum.photos/id/1067/300/300',
        title: 'Cityscapes Collection',
        tagTitle: 'Cityscapes',
        tagColour: '#7baab5',
        index: 3,
    },
]
