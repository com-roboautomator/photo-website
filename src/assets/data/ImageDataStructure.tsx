export default interface collection {
    key: string
    title: string
    index: number

    images?: image[]
    titleImage?: number

    tagTitle?: string
    tagColour?: string
}

export interface image {
    key: string
    url: string
    title: string
    index: number
}

export const placeholder: image = {
    key: '44525db2-7e1d-4201-a07c-b7d11f686bf7',
    url: 'https://picsum.photos/id/250/500',
    title: 'Placeholder',
    index: 0,
}
