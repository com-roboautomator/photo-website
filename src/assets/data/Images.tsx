import collection from './ImageDataStructure'
import { v4 as uuid } from 'uuid'

export const home: collection[] = [
    {
        key: 'befee15c-c770-4450-b71f-f20f820b3549',
        // url:
        //     'https://i.picsum.photos/id/1018/3914/2935.jpg?hmac=3N43cQcvTE8NItexePvXvYBrAoGbRssNMpuvuWlwMKg',
        title: 'Nature Collection',
        tagTitle: 'Nature',
        tagColour: '#47742c',
        index: 0,

        images: [
            {
                key: '9b0ba98c-f06b-443b-a96a-e480bafe3b84',
                url: 'https://picsum.photos/id/1018/800',
                title: 'Nature',
                index: 0,
            },
        ],
    },

    {
        key: '16bdf4fe-692d-4e11-8fd4-9ea37e3fc34d',
        // url:
        //     'https://i.picsum.photos/id/101/2621/1747.jpg?hmac=cu15YGotS0gIYdBbR1he5NtBLZAAY6aIY5AbORRAngs',
        title: 'Urban Collection',
        tagTitle: 'Urban',
        index: 1,

        images: [
            {
                key: 'e60e7a56-6ac5-4a81-b9df-d8e4f5d754b6',
                url: 'https://picsum.photos/id/101/200',
                title: 'Nature',
                index: 0,
            },
        ],
    },

    {
        key: '2c615617-6372-4042-bce5-a97910a51b5a',
        // url:
        //     'https://i.picsum.photos/id/1024/1920/1280.jpg?hmac=-PIpG7j_fRwN8Qtfnsc3M8-kC3yb0XYOBfVzlPSuVII',
        title: 'Animal Collection',
        tagTitle: 'Nature',
        tagColour: '',
        index: 2,

        images: [
            {
                key: 'bb8c95bf-be11-4fcd-a6a1-376db689bf93',
                url: 'https://picsum.photos/id/1024/200',
                title: 'Nature',
                index: 0,
            },
        ],
    },

    {
        key: '3961f578-14ea-46aa-9b74-7f908fcd532b',
        // url:
        //     'https://i.picsum.photos/id/111/4400/2656.jpg?hmac=leq8lj40D6cqFq5M_NLXkMYtV-30TtOOnzklhjPaAAQ',
        title: 'Mechanical Collection',
        tagTitle: 'Technology',
        tagColour: '#50b1d6',
        index: 3,

        images: [
            {
                key: '7215bc0b-d866-496c-8b8a-c0022d9391ef',
                url: 'https://picsum.photos/id/111/200',
                title: 'Nature',
                index: 0,
            },
        ],
    },

    {
        key: '37f92b19-2ea1-4742-b97b-81ed97e2e9c3',
        // url:
        //     'https://i.picsum.photos/id/158/4836/3224.jpg?hmac=Gu_3j3HxZgR74iw1sV0wcwlnSZSeCi7zDWLcjblOp_c',
        title: 'Party Collection',
        index: 4,

        images: [
            {
                key: '65fcf4f0-cc44-4c1c-86f9-9a14c8dcaea2',
                url: 'https://picsum.photos/id/158/200',
                title: 'Nature',
                index: 0,
            },
        ],
    },
    {
        key: '6dcdba12-f48c-46cf-b1f6-07e8d1c5ed2b',
        // url:
        //     'https://i.picsum.photos/id/1018/3914/2935.jpg?hmac=3N43cQcvTE8NItexePvXvYBrAoGbRssNMpuvuWlwMKg',
        title: 'Second Nature Collection',
        index: 5,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1018/200',
                title: 'Nature',
                index: 0,
            },
        ],
    },

    {
        key: uuid(),
        // url:
        //     'https://i.picsum.photos/id/101/2621/1747.jpg?hmac=cu15YGotS0gIYdBbR1he5NtBLZAAY6aIY5AbORRAngs',
        title: 'Second Urban Collection',
        index: 6,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/101/200',
                title: 'Nature',
                index: 0,
            },
        ],
    },

    {
        key: uuid(),
        // url:
        //     'https://i.picsum.photos/id/1024/1920/1280.jpg?hmac=-PIpG7j_fRwN8Qtfnsc3M8-kC3yb0XYOBfVzlPSuVII',
        title: 'Second Animal Collection',
        index: 7,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1024/200',
                title: 'Nature',
                index: 0,
            },
        ],
    },

    {
        key: uuid(),
        // url:
        //     'https://i.picsum.photos/id/111/4400/2656.jpg?hmac=leq8lj40D6cqFq5M_NLXkMYtV-30TtOOnzklhjPaAAQ',
        title: 'Second Mechanical Collection',
        index: 8,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/111/200',
                title: 'Nature',
                index: 0,
            },
        ],
    },

    {
        key: uuid(),
        // url:
        //     'https://i.picsum.photos/id/158/4836/3224.jpg?hmac=Gu_3j3HxZgR74iw1sV0wcwlnSZSeCi7zDWLcjblOp_c',
        title: 'Second Party Collection',
        index: 9,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/158/200',
                title: 'Nature',
                index: 0,
            },
        ],
    },
    {
        key: uuid(),
        // url:
        //     'https://i.picsum.photos/id/1013/4256/2832.jpg?hmac=UmYgZfqY_sNtHdug0Gd73bHFyf1VvzFWzPXSr5VTnDA',
        title: 'Wedding Collection',
        index: 10,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1013/200',
                title: 'Nature',
                index: 0,
            },
        ],
    },
    {
        key: uuid(),
        // url:
        //     'https://i.picsum.photos/id/1023/3955/2094.jpg?hmac=AW_7mARdoPWuI7sr6SG8t-2fScyyewuNscwMWtQRawU',
        title: 'Sport Collection',
        index: 11,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1023/200',
                title: 'Nature',
                index: 0,
            },
        ],
    },
    {
        key: uuid(),
        // url:
        //     'https://i.picsum.photos/id/375/5184/3456.jpg?hmac=3OUWWnSmq1CUXU7cmTnctSvhQYvyME_osftkbJynX04',
        title: 'People Collection',
        index: 12,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/375/200',
                title: 'Nature',
                index: 0,
            },
        ],
    },
]

export const natureImages: collection[] = [
    {
        key: '017a317f-8285-4cfb-a4e7-668d85721b76',
        title: 'Animal Collection',
        titleImage: 0,
        tagTitle: 'Nature',
        tagColour: '#47742c',
        index: 0,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1003/500',
                title: 'Deer',
                index: 0,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1020/500',
                title: 'Bear & Cub',
                index: 1,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1024/500',
                title: 'Buzzard',
                index: 2,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1025/500',
                title: 'Pug',
                index: 3,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1074/500',
                title: 'Lion',
                index: 4,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1084/500',
                title: 'Walrus',
                index: 5,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/169/500',
                title: 'Dogs',
                index: 6,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/200/500',
                title: 'Moo Cow',
                index: 7,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/237/500',
                title: 'Dog',
                index: 8,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/244/500',
                title: 'Heron',
                index: 9,
            },
        ],
    },

    {
        key: 'b97bf8c2-6ab5-45e5-b363-dace717b8ed5',
        title: 'Landscape Collection',
        titleImage: 1,
        tagTitle: 'Nature',
        tagColour: '#47742c',
        index: 1,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/256/500',
                title: 'Landscape #1',
                index: 0,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/261/500',
                title: 'Sand',
                index: 1,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/277/500',
                title: 'Grass',
                index: 2,
            },
        ],
    },

    {
        key: '12352dd6-255c-4e38-9230-a96723bf9c5d',
        title: 'Romance Collection',
        tagTitle: 'Life',
        tagColour: '#7baab5',
        index: 2,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1004/500',
                title: 'People',
                index: 0,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/336/500',
                title: 'Locks',
                index: 1,
            },
        ],
    },

    {
        key: 'ffdf5e29-408c-44e9-a453-78ba0adb91f5',
        title: 'Extreme Sporting Collection',
        tagTitle: 'Life',
        tagColour: '#7baab5',
        index: 3,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/328/500',
                title: 'Skiing',
                index: 0,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/386/500',
                title: 'Hiking',
                index: 1,
            },
        ],
    },

    {
        key: 'a2cb8bb1-0d7e-445d-b977-be97c616aed3',
        title: 'Party Collection',
        tagTitle: 'Life',
        tagColour: '#7baab5',
        index: 4,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/158/500',
                title: 'Party',
                index: 0,
            },
        ],
    },
]

export const urbanImages: collection[] = [
    {
        key: '20716a73-c26a-41b1-b309-93f219f14bd2',
        titleImage: 4,
        title: 'Buildings Collection',
        tagTitle: 'Architecture',
        tagColour: '#5f5f5f',
        index: 0,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1031/300/300',
                title: 'Skyscraper',
                index: 0,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1029/300/300',
                title: 'New York',
                index: 1,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1040/300/300',
                title: 'Castle',
                index: 2,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1047/300/300',
                title: 'Alley',
                index: 3,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1067/300/300',
                title: 'City View',
                index: 4,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/164/300/300',
                title: 'Lake Side View',
                index: 5,
            },
        ],
    },

    {
        key: '7216f8b0-55eb-4a2c-949c-e46948ecb1d8',
        titleImage: 0,
        title: 'Transport Collection',
        tagTitle: 'Cityscapes',
        tagColour: '#7baab5',
        index: 1,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/364/300/300',
                title: 'Trains',
                index: 0,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1033/300/300',
                title: 'Station',
                index: 1,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1063/300/300',
                title: 'Road Through Trees',
                index: 2,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1077/300/300',
                title: 'Cycling',
                index: 3,
            },
        ],
    },

    {
        key: 'af28d1d2-fcd7-4138-9090-a0e6ac2a38fe',
        titleImage: 2,
        title: 'Historic Building Collection',
        tagTitle: 'Architecture',
        tagColour: '#5f5f5f',
        index: 2,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/122/300/300',
                title: "Looks a little like London, but fairly sure it isn't",
                index: 0,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1040/300/300',
                title: 'Castle',
                index: 1,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/134/300/300',
                title: 'Bridge',
                index: 2,
            },
        ],
    },

    {
        key: '5e5e0130-cd61-43b1-9275-0020dba39cc7',
        titleImage: 5,
        title: 'Cityscapes Collection',
        tagTitle: 'Cityscapes',
        tagColour: '#7baab5',
        index: 3,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/142/300/300',
                title: 'Cityscape #1',
                index: 0,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/164/300/300',
                title: 'Cityscape #2',
                index: 1,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/178/300/300',
                title: 'Through window pane',
                index: 2,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/188/300/300',
                title: 'Cityscape #4',
                index: 3,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/214/300/300',
                title: 'Cityscape #5',
                index: 4,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/221/300/300',
                title: 'Cityscape #6',
                index: 0,
            },
        ],
    },
]

// this is temporary and will be replaced with a more robust API
export function findImageById(id: string) {
    for (var i = 0; i < natureImages.length; i++) {
        const collection = natureImages[i].images
        for (var j = 0; collection && collection.length > j; j++) {
            console.log(collection[j].key + " : " + id)

            if (id === collection[j].key) return collection[j]
        }
    }
    for (var k = 0; k < home.length; k++) {
        const collection = home[k].images
        for (var l = 0; collection && collection.length < l; l++) {
            if (id === collection[l].key) return collection[l]
        }
    }

    for (var m = 0; m < urbanImages.length; m++) {
        const collection = urbanImages[m].images
        for (var n = 0; collection && collection.length < n; n++) {
            if (id === collection[n].key) return collection[n]
        }
    }
}

export function findCollectionById(id: string) {
    for (var i = 0; i < natureImages.length; i++) {
        const collection = natureImages[i]
        if (id === collection.key) return collection
    }
    for (var k = 0; k < urbanImages.length; k++) {
        const collection = urbanImages[k]
        if (id === collection.key) return collection
    }
    for (var m = 0; m < home.length; m++) {
        const collection = home[m]
        if (id === collection.key) return collection
    }
}