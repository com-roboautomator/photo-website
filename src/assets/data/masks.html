<mask id="mask">
    <ellipse cx="50%" cy="50%" rx="25%" ry="25%" fill="white" filter="url(#filter)"></ellipse>
</mask>