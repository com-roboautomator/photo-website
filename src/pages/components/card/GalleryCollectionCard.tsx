import {createStyles, Link, withStyles, WithStyles} from '@material-ui/core'
import React from 'react'
import '../../../assets/css/OpenSans-Light.ttf'
import imageIcon from '../../../assets/images/icons/collection-image-o.png'
import CollectionTag from './CollectionTag'

interface GalleryCollectionCardProps extends WithStyles<typeof styles> {
    key?: string
    coverSrc: string
    title: string
    tagTitle?: string
    tagColour?: string
    selected: boolean
    height: number
    onClick?: () => void
}

class GalleryCollectionCard extends React.Component<GalleryCollectionCardProps> {
    render() {
        const classes = this.props.classes
        return (
           
                <div
                    data-testid="Collection-Card-Spacing"
                    className={
                        this.props.selected
                            ? classes.spacing_selected
                            : classes.spacing_unselected
                    }>
                    <div className={classes.container}>
                        <Link href={this.props.coverSrc} target="_blank">
                            <div className={classes.image_wrapper}>
                                <img
                                    data-testid="Collection-Card-Image"
                                    className={classes.image}
                                    src={this.props.coverSrc}
                                    alt=""
                                    width="350"
                                    height={this.props.height - 20}
                                />
                            </div>
                        </Link>
                    </div>
                </div>
        )
    }
}

const styles = () =>
    createStyles({
        spacing_selected: {
            pointerEvents: 'auto',
            userSelect: 'none',
            position: 'relative',
            zIndex: 99,
            WebkitTransition: 'opacity 0.7s, transform 0.4s, margin 0.4s',
            transform: 'scale(1)',
            opacity: '100%',
            flex: 1,
        },
        spacing_unselected: {
            pointerEvents: 'auto',
            userSelect: 'none',
            flex: 1,
            WebkitTransition: 'opacity 0.7s, transform 0.4s, margin 0.4s',
            transform: 'scale(0.8)',
            opacity: '90%',
        },
        container: {
            //width: 350,
            //borderTopLeftRadius: '25px',
            //borderTopRightRadius: '25px',
            //borderBottomLeftRadius: '25px',
            //borderBottomRightRadius: '25px',
            boxShadow: '1px 2px 10px rgba(0, 0, 0, 0.6)',
        },
        image: {
            //borderTopLeftRadius: '25px',
            //borderTopRightRadius: '25px',
        },
        image_wrapper: {
            position: 'relative',
            borderTopLeftRadius: '25px',
            borderTopRightRadius: '25px',
            display: 'flex',
            flex: 1,
            flexDirection: 'row',
        },
        text_wrapper: {
            position: 'relative',
            zIndex: 99,
            display: 'flex',
            flex: 1,
            flexDirection: 'row',
            fontSize: '100%',
            color: 'black',
            //borderBottomLeftRadius: '25px',
            //borderBottomRightRadius: '25px',
            borderTop: '1px solid gray',
            height: '45px',
        },
        icon: {
            paddingLeft: '20px',
            alignSelf: 'center',
        },
        text: {
            fontFamily: 'Open-Sans-Regular',
            fontStyle: 'normal',
            textAlign: 'left',
            position: 'relative',
            paddingLeft: '10px',
            width: '75%',
            alignSelf: 'center',
        },
        tag: {
            position: 'relative',
            paddingLeft: '10px',
            paddingRight: '20px',
            width: '25%',
            alignSelf: 'center',
        },
    })

export default withStyles(styles)(GalleryCollectionCard)
