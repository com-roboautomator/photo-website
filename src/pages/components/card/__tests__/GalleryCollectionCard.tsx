import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import GalleryCollectionCard from '../GalleryCollectionCard'
import { natureImages } from '../../../../assets/data/Images'

describe('Gallery Collection Card', () => {
    it('should match snapshot', () => {
        const ImageSliderRender = renderer
            .create(
                <GalleryCollectionCard
                    key="test-card-key"
                    height={550}
                    selected={0}
                    collection={natureImages[0]}
                />
            )
            .toJSON()
        expect(ImageSliderRender).toMatchSnapshot()
    })

    it('should render correctly', () => {
        const { getByTestId } = render(
            <GalleryCollectionCard
                key="test-card-key"
                height={550}
                selected={0}
                collection={natureImages[0]}
            />
        )

        expect(getByTestId).toBeTruthy()
        expect(getByTestId('Collection-Card-Spacing')).toBeInTheDocument()
        expect(getByTestId('Collection-Card-Link')).toBeInTheDocument()
        expect(getByTestId('Collection-Card-Image')).toBeInTheDocument()
        expect(getByTestId('Gallery-Collection-Card-Text')).toBeInTheDocument()
    })

    it('should render target image correctly', () => {
        const { getByTestId } = render(
            <GalleryCollectionCard
                key="test-card-key"
                height={550}
                selected={0}
                collection={natureImages[0]}
            />
        )

        expect(getByTestId).toBeTruthy()
        expect(getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'transform: scale(1)'
        )
    })

    it('should render immediate background image correctly', () => {
        const { getByTestId } = render(
            <GalleryCollectionCard
                key="test-card-key"
                height={550}
                selected={1}
                collection={natureImages[0]}
            />
        )

        expect(getByTestId).toBeTruthy()
        expect(getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'transform: scale(0.8)'
        )
        expect(getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'marginLeft: -100px'
        )
        expect(getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'marginRight: -100px'
        )
        expect(getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'filter: blur(1px)'
        )
    })

    it('should render distance background image correctly', () => {
        const { getByTestId } = render(
            <GalleryCollectionCard
                key="test-card-key"
                height={550}
                selected={2}
                collection={natureImages[0]}
            />
        )

        expect(getByTestId).toBeTruthy()
        expect(getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'transform: scale(0.5)'
        )
        expect(getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'marginLeft: -100px'
        )
        expect(getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'marginRight: -100px'
        )
        expect(getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'filter: blur(4px)'
        )
    })
})
