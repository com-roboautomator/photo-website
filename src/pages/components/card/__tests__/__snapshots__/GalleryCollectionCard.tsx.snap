// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Collection Card should match snapshot 1`] = `
<div
  className="GalleryCollectionCard-spacing_selected-1"
  data-testid="Collection-Card-Spacing"
>
  <div
    className="GalleryCollectionCard-container-5"
  >
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      data-testid="Collection-Card-Link"
      href="gallery/017a317f-8285-4cfb-a4e7-668d85721b76"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="GalleryCollectionCard-image_wrapper-6"
      >
        <img
          alt=""
          data-testid="Collection-Card-Image"
          height={530}
          src="https://picsum.photos/id/1003/500"
          width="350"
        />
        <div
          className="GalleryCollectionCard-container-7"
          data-testid="Gallery-Collection-Card-Text"
        >
          <div
            className="GalleryCollectionCard-tag-8"
            data-testid="Gallery-Collection-Card-Tag"
            style={
              Object {
                "borderBottomColor": "#47742c",
              }
            }
          >
            NATURE
          </div>
          <div
            className="GalleryCollectionCard-text-9"
            data-testid="Gallery-Collection-Card-Content-Text"
          >
            Animal Collection
            <div
              className="GalleryCollectionCardItemsText-container-10"
              data-testid="Gallery-Collection-Card-Items-Text"
            >
              10
               Items
            </div>
          </div>
          <div
            className="GalleryCollectionCardViewIcon-container-11"
          >
            <div
              className="GalleryCollectionCardViewIcon-image-12"
            >
              <img
                alt="none"
                data-testid="Gallery-Collection-Card-View-Icon-Image"
                height="20"
                src="collection-image-w.png"
                width="20"
              />
            </div>
            <div
              className="GalleryCollectionCardViewIcon-text-13"
              data-testid="Gallery-Collection-Card-View-Icon-Text"
            >
              VIEW
            </div>
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;
